File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        //scan from left side and right side
        int left = 0;
        int right = numbers.length - 1;

        //while the pointers dont cross each other
        while(left<right){
            int twoSum = numbers[left] + numbers[right];

            //if sum too big decrease right pointer i
            //if sum too small increase left pointer
            //else return left and right pointer 1 indexed
            if(twoSum == target){
                return new int[]{left + 1, right + 1};
            }else if(twoSum < target){
                left++;
            }else{
                right--;
            }
        }

        return new int[] {};
    }
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        """
        since the array is sorted and guaranteed a solution
        we can use ptrs to keep track of the curr sum

        if the curr sum is greater than the target, we move the right ptr inwards, if it is less than, 
        then we can move the left ptr inwrads
        """

        l,r = 0,len(numbers)-1

        # find curr sum
        while l < r:
            curr_sum = numbers[l] + numbers[r]

            if curr_sum == target:
                return [l+1,r+1]
            elif curr_sum < target:
                l += 1
            else:
                r -= 1

    